bazel/linux/defs.bzl

```diff
@@ -261,37 +329,68 @@ def kernel_module(*args, **kwargs):
       kernels: list of kernel (same as above). kernel_module will instantiate multiple
             kernel_module_rule, one per kernel, and ensure they all build in parallel.
     """
+
     if "makefile" not in kwargs:
         kwargs["makefile"] = "Makefile"
 
     if "srcs" not in kwargs:
         include = ["**/*.c", "**/*.h", kwargs["makefile"]]
         kwargs["srcs"] = native.glob(include = include, exclude = BUILD_LEFTOVERS, allow_empty = False)
 
-    module = kwargs.get("module", kwargs["name"])
-    if not module.endswith(".ko"):
-        module = module + ".ko"
-    kwargs["module"] = module
+    if "module" in kwargs:
+        module = kwargs["module"]
+        kwargs.pop("module")
+    else:
+        module = kwargs["name"]
```

Comment:
or if you want to support both having a single module or a list of modules...
```
kwargs.setdefault("modules", [kwargs.pop("module", kwargs["name"])])
```
although this will silently override modules with module if both are specified.